---
title: "Juan Project"
output: 
  html_document:
    toc: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
  encoding=encoding,
  output_file=file.path(dirname(input_file), out_dir, 'index.html'))})

---

```{r packages, message=FALSE, warning=FALSE, echo=FALSE}
load("workspace.RData")
#mywd
#knit: (function(input_file, encoding) {
#  out_dir <- 'docs';
#  rmarkdown::render(input_file,
# encoding=encoding,
# output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
#https://resources.github.com/whitepapers/github-and-rstudio/ #info on github page creation
setwd("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration")
#mypackages
library(ggrepel)
library(qiime2R)
library(tidyverse)
library(stringr)
library(ggplot2)
theme_set(theme_bw())  
library(vegan)
library(ggpubr)
library(RColorBrewer)
library(phyloseq)
library(psych)
library(reshape2)
library(ggfortify)
library(factoextra)
library(FactoMineR)
library(maigesPack)
library(knitr)
library(microbiome)
library(SpiecEasi)
library(bookdown)
library(tinytex)
library(jakR)
library(kableExtra)
options(kableExtra.auto_format = FALSE)
```


# Metadata and phyloseq objects {#metadata}
```{r phyloseq, warning=FALSE, message=FALSE, echo=FALSE, cache = FALSE}
#https://forum.qiime2.org/t/tutorial-integrating-qiime2-and-r-for-data-visualization-and-analysis-using-qiime2r/4121?u=jbisanz
metadata<-read_tsv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/metadata.tsv") #requires tidyverse
metadata2 <- metadata[c(2:67),]
#defining factors
metadata2$Soiltypes <- factor(metadata2$Soiltypes, levels=c("Sodosol", "Vertisol","PLcomparison"))
metadata2$Treatments <- factor(metadata2$Treatments, levels=c("Control", "NPKS", "PL/N", "PL", "Straw+NPKS",  "no plant no amendment", "PLharvest","PLorig"), labels = c("Control","NPKS", "Poultry litter/N", "Poultry litter", "NPKS/straw","No plant No amendment","PLharvest","PLorig"))
metadata2$AggregateSize <- factor(metadata2$AggregateSize, levels=c(">2 mm", "0.25-2 mm", "PL"))

# create phyloseq object
SVs      <-   qiime2R::read_qza("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/qiime2_data_200317/feature_table.qza")
#SVs      <-   qiime2R::read_qza("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/qiime2_data_200317/feature_table_min2splsmin10freq.qza")
taxonomy <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/qiime2_data_200317/taxonomy_silva.tsv") %>% filter(Taxon != "categorical")  %>% rename(Feature.ID = `Feature ID`)
taxtable <- taxonomy %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) 
# did not do the tree yet for the new set filtered on 17March20
#tree <- read_qza("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/qiime2_data/insertion-tree.qza")

physeqB <- phyloseq(
  otu_table(SVs$data, taxa_are_rows = T), 
  #phy_tree(tree$data), 
  tax_table(as.data.frame(taxtable) %>% dplyr::select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata2 %>% as.data.frame() %>% column_to_rownames("#SampleID")))

#adding diversity measures (with ps object) to metadata and re-create ps object 
#here a decision has to be made on sample size. Look at feature-table.qzv to see how many samples are lost with x sample size or how many features are lost with minimum sample size. This is used ONLY for diversity metrics (not differentially abundance anaylyis which I filter separately. So I generally use the minimum samples size if different between min/max is within 10fold. 
print(paste("The minimum number of sequences in this data set is",min(colSums(otu_table(physeqB))),"and the maximum number of sequences in this data set is",max(colSums(otu_table(physeqB))) ))
#Create rarefied physeq object
rare_Bac <- rarefy_even_depth(physeqB, sample.size = min(colSums(otu_table(physeqB))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.div <- estimate_richness(rare_Bac) #do diversity measurements and add to metadata
ps.div$`#SampleID` <- sample_names(physeqB)

meta.df <- data.frame(sample_data(physeqB))
meta.df <- meta.df %>% rownames_to_column("#SampleID") %>% 
  left_join(ps.div, by = "#SampleID") %>% column_to_rownames("#SampleID")
meta.df$pielou_ev <- ps.div$Shannon/log(ps.div$Observed)      #Pielou eveness
physeqB@sam_data <- sample_data(meta.df)

physeqB
#kable(head(sample_data(physeqB)))
#head(otu_table(physeqB))
#head(tax_table(physeqB))
kable(meta.df[c(1:5),c(1:12)], booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")


# now for the pathway file
pathway.pred <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/path_abun_unstrat.tsv", col_names = TRUE)  %>% column_to_rownames("pathway")
pw.names <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Milestone2_Field_Survey/M2_Bacteria/Picrust/metacyc_pathway_info_full.tsv") %>% column_to_rownames("Id")  %>% dplyr::select(-ALTERNATIVE) %>% dplyr::select(-`SUPER-PATHWAYS`)%>% dplyr::select(-NAMES) %>% dplyr::select(-CLASS2)

physeq.pw <- phyloseq(otu_table(
             pathway.pred, taxa_are_rows = T), 
              tax_table(as.data.frame(pw.names) %>% as.matrix()),
             sample_data(metadata2 %>% as.data.frame() %>% column_to_rownames("#SampleID"))) 
taxtable <- data.frame(`COMMON.NAME` = (tax_table(physeq.pw)@.Data[,c("COMMON-NAME")]))
taxa_names(physeq.pw) <- taxtable$COMMON.NAME



```

# Data inspection and filtering {#inspec}
```{r inspec, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
# all features
otus.table <- as.data.frame(otu_table(physeqB))
otus.table <- otus.table %>% rownames_to_column(var = "#SampleID") %>% arrange(desc(rowSums(otus.table))) %>% column_to_rownames(var = "#SampleID")
#filter 
#minimum of reads per feature
physeqB.flt = prune_taxa(taxa_sums(physeqB) >= 10, physeqB) #minimum reads per feature
physeqB.flt = filter_taxa(physeqB.flt , function(x) sum(x > 0) > (0.03*length(x)), TRUE) #minimum presense in x% of samples
#filter any phyla that have not been classified i.e. are "NA"
physeqB.flt = subset_taxa(physeqB.flt, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))

#rarefaction curve
rarecurve(t(otu_table(physeqB.flt)), step=200, sample = min(colSums(otu_table(physeqB.flt))), label = FALSE) 

#taxatables for reference
taxtableB.fltr <- data.frame(phyloseq::tax_table(physeqB.flt)@.Data)
taxtableB.fltr <- taxtableB.fltr %>% rownames_to_column(var = "Id")
# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(physeqB.flt))
summary(sample_sum_df$sum)

#histograms
# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "indianred") +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank())

# barplot with red line through median value
dat2 <- t(otu_table(physeqB.flt)) %>% melt(.) %>% arrange(desc(value))
ggbarplot(data = dat2, x = "Var2", y = "value",xlab = "Filtered", order = unique(dat2$Var2)) + geom_hline(yintercept=median(rowSums(otu_table(physeqB.flt))), color="red")

summary(taxa_sums(physeqB.flt))
print(paste("The number of reads and ASVs left after filtering is",round(sum(taxa_sums(physeqB.flt)) / sum(taxa_sums(physeqB))*100,2),"% and",round(ntaxa(physeqB.flt) / ntaxa(physeqB) *100,2),"%, a change from",round(sum(taxa_sums(physeqB))),"to",round(sum(taxa_sums(physeqB.flt))), "reads and", ntaxa(physeqB),"to",ntaxa(physeqB.flt),"ASVs. The median changed from", median(rowSums(otu_table(physeqB))),"to",median(rowSums(otu_table(physeqB.flt))),"reads per ASV."))

otutable <- otu_table(physeqB.flt) %>% as.data.frame() %>% rownames_to_column("Feature.ID") %>% left_join(taxonomy) %>% dplyr::select( - Confidence, -Feature.ID) %>% dplyr::select(Taxon, everything())
#write.csv(t(otutable), "otu.table_Juan.csv", row.names = TRUE)
```
go [back to top](#metadata) 

# Taxa barplots
## Comparison by soil
```{r barsoil, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE}
#create plotting tables
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "Soiltypes")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#use this for plots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
#barplot
barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered OTUs", legend = "right") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
```
go [back to top](#metadata) 

## Comparison by Soil, treatment and aggregate size
```{r soiltreament, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE}
#Phylum level
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )

#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.relsoil1 <- barplot.rel %>% filter(Soiltypes != "PLcomparison") %>%  ggbarplot(x = "AggregateSize", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soiltypes","Treatments")) +  theme(plot.title = element_text(size = 10)) + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1
```

```{r soiltreamentPL, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
# Poultry litter comparison
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.relsoil1 <- barplot.rel %>% filter(Soiltypes == "PLcomparison") %>%  ggbarplot(x = "Treatments", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soiltypes","AggregateSize")) +  theme(plot.title = element_text(size = 10)) + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1
```


```{r sodtsigdiff, include = TRUE, echo=FALSE, message=FALSE, warning=FALSE, fig.asp=0.8, fig.width= 15, cache=TRUE,eval=FALSE}

#Soil1 data
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soiltypes == "Sodosol", physeqB.flt)

#Soil1
#wilcoxon test
barplot.rel <- microbiome::transform(physeqBSoil1, "clr")
barplot.rel.melt <- psmelt(barplot.rel) %>%  dplyr::arrange(desc(Phylum))
wilcoxotest <-  barplot.rel.melt %>% ggpubr::compare_means(Abundance ~ Treatments, data = ., group.by = "Genus", method = "wilcox.test", paired = FALSE, ref.group = "Control", symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% dplyr::filter(p.adj <= 0.001)
#figvec <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/filtervecforgraph.csv", header = TRUE)

vecS <- c("D_5__Acidibacter", "D_5__Actinoallomurus","D_5__Actinoplanes", "D_5__Anaeromyxobacter", "D_5__Aquisphaera", "D_5__Bacillus", "D_5__Bdellovibrio","D_5__Bryobacter", "D_5__Candidatus Koribacter", "D_5__Candidatus Solibacter", "D_5__Cellvibrio", "D_5__Chthoniobacter", "D_5__Devosia", "D_5__Ellin6067","D_5__Flavisolibacter", "D_5__Flavobacterium", "D_5__Gemmatimonas", "D_5__Gemmatirosa","D_5__Haliangium","D_5__Isosphaera", "D_5__LD29", "D_5__Ktedonobacter", "D_5__Luteolibacter", "D_5__Microbacterium","D_5__Micromonospora", "D_5__Nakamurella", "D_5__Nannocystis", "D_5__Niastella", "D_5__Nitrolancea", "D_5__Nocardioides", "D_5__Novibacillus", "D_5__Oceanobacillus", "D_5__Ohtaekwangia", "D_5__Oligoflexus", "D_5__Opitutus", "D_5__Paenibacillus", "D_5__Pajaroellobacter", "D_5__Pedobacter", "D_5__Peredibacter","D_5__Pirellula", "D_5__Reyranella", "D_5__Rhodanobacter", "D_5__Rhodopirellula", "D_5__Rubrobacter", "D_5__Segetibacter", "D_5__Singulisphaera", "D_5__Sphingobacterium", "D_5__Sphingomonas", "D_5__Streptomyces", "D_5__Truepera", "D_5__Virgibacillus", "D_5__Vulgatibacter")

wilcoxotest <- wilcoxotest %>% dplyr::filter(Genus %in% vecS) 
kable(wilcoxotest,booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")

#barplot
barplot.rel <- phyloseq::transform_sample_counts(physeqBSoil1, function(x) x / sum(x) )
subs <- psmelt(barplot.rel) %>% dplyr::filter(Genus %in% vecS) 
barplot.relsoil1 <- subs  %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + theme(axis.title.x=element_blank(),             axis.text.x=element_blank(), axis.ticks.x=element_blank()) # theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
barplot.relsoil1 +  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())


```

```{r sodtsigdiffbarplot, echo=FALSE, message=FALSE, fig.width=15,  fig.asp=1, warning=FALSE, include = FALSE, eval=FALSE, cache=TRUE}
#barplot

figvec <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/juancollaboration/filtervecforgraph.csv", header = TRUE)
figvec <- as.vector(figvec$x)
barplot.rel <- phyloseq::transform_sample_counts(physeqBSoil1, function(x) x / sum(x) )

subs <- psmelt(barplot.rel) %>% dplyr::filter(Genus %in% figvec)
barplot.relsoil1 <- subs  %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + theme(axis.title.x=element_blank(),             axis.text.x=element_blank(), axis.ticks.x=element_blank()) # theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
barplot.relsoil1 
```

 
```{r vertsigdiff, echo=FALSE, fig.width=15,  fig.asp=0.6, message=FALSE, warning=FALSE, cache=TRUE, eval = FALSE}
#Soil2 data
physeqBSoil2 <-  prune_samples(sample_data(physeqB.flt)$Soiltypes == "Vertisol", physeqB.flt)

#Soil1
#wilcoxon test
barplot.rel <- microbiome::transform(physeqBSoil2, "clr")
barplot.rel.melt <- psmelt(barplot.rel) %>%  arrange(desc(Phylum))
wilcoxotest <-  barplot.rel.melt %>% compare_means(Abundance ~ Treatments, data = ., group.by = "Genus", method = "wilcox.test", paired = FALSE, ref.group = "Control", symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% filter(p.adj <= 0.001)
kable(wilcoxotest %>% filter(Genus != ""), booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")


#barplot
#vec <- unique(wilcoxotest %>% filter(Genus != "") %>% dplyr::select(Genus))  
vec <- c("D_5__Bdellovibrio", "D_5__Candidatus Solibacter","D_5__Candidatus Udaeobacter","D_5__Conexibacter", "D_5__Devosia" , "D_5__Flavisolibacter","D_5__Flavobacterium","D_5__Gemmatimonas", "D_5__Nitrolancea","D_5__Mucilaginibacter", "D_5__Ohtaekwangia", "D_5__Peredibacter", "D_5__Pirellula", "D_5__Pseudomonas" , "D_5__Pseudogracilibacillus")
barplot.rel <- phyloseq::transform_sample_counts(physeqBSoil2, function(x) x / sum(x) )
subs <- psmelt(barplot.rel) %>% dplyr::filter(Genus %in% vec)

barplot.relsoil2 <- subs  %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") # theme(axis.text.x = element_text(angle = 45, hjust = 1))

barplot.relsoil2 

```

go [back to top](#metadata)  



# Alpha diversity {#alphadiversity}
## by soil
```{r alphasoil, echo=FALSE, fig.width=6, message=FALSE, warning=FALSE, cache=TRUE}
as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed", "InvSimpson", "Shannon", "pielou_ev"))  %>%
ggboxplot(., x = "Soiltypes", y = "value", add = "jitter", facet.by = "variable", scales = "free", ylab = "alpha diversity") +
  stat_compare_means( method = "kruskal") 
```
  
go [back to top](#metadata) 


## Comparison by treatments
```{r alphaCO2, echo=FALSE, fig.asp=1.5, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE}
p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>% ggboxplot(., x = "Treatments", y = "value", add = "jitter", facet.by = c("Soiltypes","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox", ref.group = "Control", label = "p.signif") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson")) %>% ggboxplot(., x = "Treatments", y = "value", add = "jitter", facet.by = c("Soiltypes","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox", ref.group = "Control", label = "p.signif") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))  %>% ggboxplot(., x = "Treatments", y = "value", add = "jitter", facet.by = c("Soiltypes","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox", ref.group = "Control", label = "p.signif") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))  %>% ggboxplot(., x = "Treatments", y = "value", add = "jitter", facet.by = c("Soiltypes","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox", ref.group = "Control", label = "p.signif") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4, nrow = 2, ncol = 2, heights  = c(0.6,1))
x = dev.off()
gg1

```
  
go [back to top](#metadata) 



# Beta diversity 

## NMDS Bray
### Sodosol and Vertisol (n = 6 per treatment)
Data was filtered to ASVs with a minimum of 10 reads and then rarefied to an even depth before analysis
```{r nmdsSod, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)
symbolsize = 3

#Soil1 data
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soiltype == "Sodosol", physeqB.flt)

ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
#ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds1 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Sodosol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) +
  geom_point(size=symbolsize)

#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")

```
  
```{r nmdsVert, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)
colorsvertisol <- c("tomato", "burlywood3", "forestgreen","hotpink")

#Soil1 data
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soiltype == "Vertisol", physeqB.flt)

ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
#ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds2 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Vertisol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) + scale_shape_manual(values=c(16, 17, 15,8)) + scale_colour_manual(values=c(colorsvertisol)) +
  geom_point(size=symbolsize)
#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
```

```{r nmdsplot, fig.width= 6, fig.asp=1.2, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE,  fig.cap = "Beta diversity comparisons by non-metric multidimensional scaling (NMDS) among different amendment treatments in the Sodosol (a) and Vertosol (b) subsoils."}
ggpubr::ggarrange(nmds1, nmds2, ncol = 1, labels = "auto") 
```

go [back to top](#metadata) 

#### Aggregate size >2 mm only (n = 3 per treatment)
```{r nmdsSod2mm, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)


#Soil1 data
physeqBSoil1 <- prune_samples(sample_data(physeqB.flt)$Soiltype == "Sodosol", physeqB.flt)
physeqBSoil1 <- prune_samples(sample_data(physeqBSoil1)$AggregateSize == ">2 mm", physeqBSoil1)

ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
# ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds1 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Sodosol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) +
  geom_point(size=symbolsize)

#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")

```
  
```{r nmdsVert2mm, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)


#Soil1 data
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soiltype == "Vertisol", physeqB.flt)
physeqBSoil1 <-  prune_samples(sample_data(physeqBSoil1)$AggregateSize == ">2 mm", physeqBSoil1)

ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
#ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds2 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Vertisol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) + scale_shape_manual(values=c(16, 17, 15,8)) + scale_colour_manual(values=c(colorsvertisol))+
  geom_point(size=symbolsize)
#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
```

```{r nmdsplot2mm, fig.width= 6, fig.asp=1.2, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE,  fig.cap = "Beta diversity comparisons by non-metric multidimensional scaling (NMDS) among different amendment treatments in the Sodosol (a) and Vertosol (b) subsoils."}
ggpubr::ggarrange(nmds1, nmds2, ncol = 1, labels = "auto") 
```

go [back to top](#metadata) 

#### Aggregate size 0.25-2 mm only (n = 3 per treatment)
```{r nmdsSod025mm, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)


#Soil1 data
physeqBSoil1 <- prune_samples(sample_data(physeqB.flt)$Soiltype == "Sodosol", physeqB.flt)
physeqBSoil1 <- prune_samples(sample_data(physeqBSoil1)$AggregateSize == "0.25-2 mm", physeqBSoil1)
physeqBSoil1 = prune_taxa(taxa_sums(physeqBSoil1) >= 10, physeqBSoil1) #minimum reads per feature
ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
#ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
#df <- otu_table(physeqBSoil1)@.Data # to check the otutable


ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds1 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Sodosol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) +
  geom_point(size=symbolsize)

#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
```
  
```{r nmdsVert025mm, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE}

# fixed width of x axis
xaxis = c(-1, 1)
yaxis = c(-0.65, 0.65)


#Soil1 data
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soiltype == "Vertisol", physeqB.flt)
physeqBSoil1 <-  prune_samples(sample_data(physeqBSoil1)$AggregateSize == "0.25-2 mm", physeqBSoil1)

ps.perm <- rarefy_even_depth(physeqBSoil1, sample.size = min(colSums(otu_table(physeqBSoil1))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
#ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")
label <- round(ord_nmds$stress,3)

nmds2 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Treatments", shape = "Treatments")  + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank() ) + xlim(xaxis) + ylim(yaxis) + geom_hline(yintercept=0, linetype="dashed", color = "gray") + geom_vline(xintercept=0, linetype="dashed", color = "gray") + annotate(geom = 'text', label = paste("Bray-Curtis, stress = ", label), x = -Inf, y = -Inf, hjust = -0.05, vjust = -1, size = 3) + annotate(geom = 'text', label = paste("Vertisol"), x = -Inf, y = 0.6, hjust = -0.2, vjust = 0, size = 4) + scale_shape_manual(values=c(16, 17, 15,8) ) + scale_colour_manual(values=c(colorsvertisol)) + geom_point(size=symbolsize)
#+ ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
```

```{r nmdsplot025mm, fig.width= 6, fig.asp=1.15, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = FALSE,  fig.cap = "Beta diversity comparisons by non-metric multidimensional scaling (NMDS) among different amendment treatments in the Sodosol (a) and Vertosol (b) subsoils."}
ggpubr::ggarrange(nmds1, nmds2, ncol = 1, labels = "auto") 
```

go [back to top](#metadata) 

<br/>

# *Bacillus sp.* plot
## Sodosol 
### Kruskal test (holm adjusted) with LSD post-hoc test (excl. outlier)
```{r bacillusblot2, fig.cap = "Total read counts, relative abundances and clr transformed abundances of the genus *Bacillus* in Sodosol without plants and no amendments (control-P-A), with plants but no amendments (control-A), amended with NPKS, NPKS+straw (NPKS/straw), poultry litter (PL), poultry litter + N (PL/N). Letters represent results of post hoc nonparametric tests (Kruskal-Wallis) at an alpha level of 0.05. Treatments with the same letter are not significantly different.", echo=FALSE, fig.width=11, fig.asp = 0.35, message=FALSE, warning=FALSE, cache=TRUE, eval=TRUE, include=TRUE}
#barplot

physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)

physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes != "PLcomparison", physeqB.flt2)
#physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments != "no plant no amendment", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)

physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.03*length(x)), TRUE) 

tax_table(physeqB.flt2)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt2)[, "Genus"], "D_5__", "")

# total counts
physeqB.flt2.mlt.count <- psmelt(physeqB.flt2)
subs <- physeqB.flt2.mlt.count %>% dplyr::filter(Genus %in% c("Bacillus")) 
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
#https://cran.r-project.org/web/packages/agricolae/vignettes/tutorial.pdf
require(agricolae)
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
#groups1 <- comparison$groups %>% rownames_to_column("Treatments")
groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)
#barplot with labels
bacilluscount <-  subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" , color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Total read counts")  + theme(axis.text.x = element_text(angle = 45, hjust = 1))  + annotate("text", x = groups1$Treatments, y=c(15500,15500,15500,15500,15500,15500), label = label) + scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen"))+ theme(legend.position = "none")


#relative
physeqB.flt2.rel <- phyloseq::transform_sample_counts(physeqB.flt2, function(x) x / sum(x) )
physeqB.flt2.mlt.rel <- psmelt(physeqB.flt2.rel)
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
subs <- physeqB.flt2.mlt.rel %>% dplyr::filter(Genus %in% c("Bacillus"))
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
#groups1 <- comparison$groups %>% rownames_to_column("Treatments")
groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)

bacillusrelative <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" ,color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + annotate("text", x = groups1$Treatments, y=c(0.38,0.38,0.38,0.38,0.38,0.38), label = label) + scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen"))+ theme(legend.position = "none")


# clr
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "clr")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)
subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% c("Bacillus"))
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
#groups1 <- comparison$groups %>% rownames_to_column("Treatments")
groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)
 
bacillusclr <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" ,color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Abundances clr transformed") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + annotate("text", x = groups1$Treatments, y=c(10,10,10,10,10,10), label = label)+ scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen"))+ theme(legend.position = "none")

ggarrange(bacilluscount, bacillusrelative , bacillusclr , legend = "none", ncol = 3)

```

```{r bacillusBOX, fig.cap = "Total read counts, relative abundances and clr transformed abundances of bacillus in Sodosol amended without (control) and with NPKS, poultry litter (PL), poultry litter/N and straw/NPKS. Significance to NPKS is shown (Wilcoxon test)." ,echo=FALSE, fig.width=8, fig.asp = 0.45, eval = FALSE, include = FALSE, message=FALSE, warning=FALSE, cache=TRUE}
#barplot

#physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)

physeqB.flt2 <- prune_samples(sample_data(physeqB)$Soiltypes != "PLcomparison", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments != "no plant no amendment", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)

physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.03*length(x)), TRUE) 

tax_table(physeqB.flt2)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt2)[, "Genus"], "D_5__", "")

# total counts
physeqB.flt2.mlt.count <- psmelt(physeqB.flt2)

subs <- physeqB.flt2.mlt.count %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacilluscounts <- subs  %>% ggboxplot(data = ., x = "Treatments", y = "Abundance", add = "jitter" , fill = "Treatments", facet.by = "Genus", xlab = "", scales = "free", ylab = "Total read counts", outlier.colour="red" ) +  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#relative
physeqB.flt2.rel <- phyloseq::transform_sample_counts(physeqB.flt2, function(x) x / sum(x) )
physeqB.flt2.mlt.rel <- psmelt(physeqB.flt2.rel)

subs <- physeqB.flt2.mlt.rel %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacillusrelative <- subs  %>% ggboxplot(data = ., x = "Treatments", y = "Abundance", add = "jitter" , fill = "Treatments", facet.by = "Genus", xlab = "", scales = "free", ylab = "Relative abundances", outlier.colour="red") +  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

# clr
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "clr")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)

subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacillusclr <- subs  %>% ggboxplot(data = ., x = "Treatments", y = "Abundance", add = "jitter" , fill = "Treatments", facet.by = "Genus", xlab = "", scales = "free", ylab = "Abundances clr transformed", outlier.colour="red") +  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggarrange(bacilluscounts, bacillusrelative , bacillusclr , common.legend = TRUE, ncol = 3)
```

### Anova with Duncan multiple comparison test (excl. outlier)
```{r bacillusblot3, fig.cap = "Total read counts, relative abundances and clr transformed abundances of the genus *Bacillus* in Sodosol without plants and no amendments (control-P-A), with plants but no amendments (control-A), amended with NPKS, NPKS+straw (NPKS/straw), poultry litter (PL), poultry litter + N (PL/N). Letters represent results of post hoc parametric tests (Duncan) at an alpha level of 0.05. Treatments with the same letter are not significantly different." , echo=FALSE, fig.width=11, fig.asp = 0.35, message=FALSE, warning=FALSE, cache=TRUE, eval=TRUE, include=TRUE}
#barplot

physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)

physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes != "PLcomparison", physeqB.flt2)
#physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments != "no plant no amendment", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)

physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.03*length(x)), TRUE) 

tax_table(physeqB.flt2)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt2)[, "Genus"], "D_5__", "")

# total counts
physeqB.flt2.mlt.count <- psmelt(physeqB.flt2)
subs <- physeqB.flt2.mlt.count %>% dplyr::filter(Genus %in% c("Bacillus")) 
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
#https://cran.r-project.org/web/packages/agricolae/vignettes/tutorial.pdf
require(agricolae)
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
groups1 <- comparison$groups %>% rownames_to_column("Treatments")
#groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)
#barplot with labels
bacilluscount <-  subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" , color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Total read counts")  + theme(axis.text.x = element_text(angle = 45, hjust = 1))  + annotate("text", x = groups1$Treatments, y=c(15500,15500,15500,15500,15500,15500), label = label) + scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen"))+ theme(legend.position = "none")


#relative
physeqB.flt2.rel <- phyloseq::transform_sample_counts(physeqB.flt2, function(x) x / sum(x) )
physeqB.flt2.mlt.rel <- psmelt(physeqB.flt2.rel)
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
subs <- physeqB.flt2.mlt.rel %>% dplyr::filter(Genus %in% c("Bacillus"))
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
groups1 <- comparison$groups %>% rownames_to_column("Treatments")
#groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)

bacillusrelative <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" ,color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + annotate("text", x = groups1$Treatments, y=c(0.38,0.38,0.38,0.38,0.38,0.38), label = label) + scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen"))+ theme(legend.position = "none")


# clr
#mutliple comparison tests 
#anova test (use agricolae package), this would not be valid on non-normal, dependant data (ie. counts and relative abundance). 
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "clr")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)
subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% c("Bacillus"))
model <- aov(Abundance ~ Treatments, data = subs)
#cv.model(model)
#Model parameters: Degrees of freedom and variance of the error:
#df <- df.residual(model)
#MSerror <- deviance(model)/df
#Alternatively the Kruskal test
kr.test <- with(subs,kruskal(Abundance,Treatments,group=TRUE, p.adj="holm"))

comparison <- duncan.test(model, "Treatments", console = FALSE)
levels  <- levels(sample_data(physeqB.flt2)$Treatments) %>% as.data.frame()
groups1 <- comparison$groups %>% rownames_to_column("Treatments")
#groups1 <- kr.test$groups %>% rownames_to_column("Treatments")
groups1 <- groups1[match(levels$., groups1$Treatments),]
label <- as.vector(groups1$groups)
 
bacillusclr <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" ,color = "grey20", fill = "Treatments", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Abundances clr transformed") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + annotate("text", x = groups1$Treatments, y=c(10,10,10,10,10,10), label = label)+ scale_x_discrete(labels = c('Control-P-A','Control-A','NPKS','NPKS/straw','PL','PL/N')) + scale_fill_manual(values = c("white", "black", "red3", "darkorange1", "springgreen3", "darkolivegreen")) + theme(legend.position = "none")

ggarrange(bacilluscount, bacillusrelative , bacillusclr , legend = "none", ncol = 3)

```


## Vertisol
```{r bacillusblot4, fig.cap = "Total read counts, relative abundances and clr transformed abundances of bacillus in Vertisol amended without (control) and with NPKS and poultry litter/N. Significance to NPKS is shown (Wilcoxon test).", echo=FALSE, fig.width=8, fig.asp = 0.45, message=FALSE, warning=FALSE, cache=TRUE}
#barplot

physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes != "PLcomparison", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments != "No plant No amendment", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Vertisol", physeqB.flt2)

physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.03*length(x)), TRUE) 

tax_table(physeqB.flt2)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt2)[, "Genus"], "D_5__", "")

# total counts
physeqB.flt2.mlt.count <- psmelt(physeqB.flt2)

subs <- physeqB.flt2.mlt.count %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacilluscounts <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se" , fill = "Treatments", xlab = "", facet.by = "Genus", scales = "free", ylab = "Total read counts", width = 0.5, position = position_dodge(0.7) )  + scale_fill_manual(values = c("tomato","khaki4","mediumseagreen") )+  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#relative
physeqB.flt2.rel <- phyloseq::transform_sample_counts(physeqB.flt2, function(x) x / sum(x) )
physeqB.flt2.mlt.rel <- psmelt(physeqB.flt2.rel)

subs <- physeqB.flt2.mlt.rel %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacillusrelative <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", fill = "Treatments", xlab = "", facet.by = "Genus", scales = "free", ylab = "Relative abundances", width = 0.5, position = position_dodge(0.7))  + scale_fill_manual(values = c("tomato","khaki4","mediumseagreen") )+  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

# clr
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "clr")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)

subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% c("Bacillus")) 
bacillusclr <- subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", fill = "Treatments", xlab = "", facet.by = "Genus", scales = "free", ylab = "Abundances clr transformed", width = 0.5, position = position_dodge(0.7))  + scale_fill_manual(values = c("tomato","khaki4","mediumseagreen") )+  stat_compare_means(method = "wilcox", ref.group = "NPKS",label = "p.signif")+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggarrange(bacilluscounts, bacillusrelative , bacillusclr , common.legend = TRUE, ncol = 3)
```
go [back to top](#metadata) 


# Genera different by treatment (to be continued)
## Sodosol
```{r aldex1, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 1.7, message=FALSE, warning=FALSE, cache=TRUE}
# Aldex

#Sodosol NPK vs PL 
require(ALDEx2)
physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments %in% c("NPKS","Poultry litter"), physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 
#agglomerate to genus level
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
taxtable.flt2 <- tax_table(physeqB.flt2)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqB.flt2)$Treatment)

#otutable <- as.data.frame(phyloseq::otu_table(physeqB.flt2))
#otutable <- round(otutable)

#cond <- sample_data(physeqB.flt2)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/x.all")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/x.all")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/aldex_significant_features.csv")
aldex_significant_features0 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/aldex_significant_features.csv")  %>% arrange(desc(effect))

#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
 # kable_styling(latex_options="scale_down")

#check them out (aldex or ancome)
#physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "compositional")
#physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)
#subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% as.vector(aldex_significant_features$Genus) )
#subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Phylum", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + ggtitle("Aldex (p corrected < 0.02)")



#Sodosol NPK vs Straw+NPKS 
require(ALDEx2)
physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments %in% c("NPKS","NPKS/straw"), physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
#agglomerate to genus level
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#Filter to genera that are present in at least 2 samples. 
physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 

taxtable.flt2 <- tax_table(physeqB.flt2)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqB.flt2)$Treatment)

#otutable <- as.data.frame(phyloseq::otu_table(physeqB.flt2))
#otutable <- round(otutable)

#cond <- sample_data(physeqB.flt2)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/x.all")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/x.all")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/aldex_significant_features.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/aldex_significant_features.csv")  %>% arrange(desc(effect))

#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
#  kable_styling(latex_options="scale_down")





#Sodosol NPK vs PL/N
require(ALDEx2)
physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Soiltypes == "Sodosol", physeqB.flt2)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments %in% c("NPKS","Poultry litter/N"), physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
#agglomerate to genus level
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
#Filter to genera that are present in at least 2 samples. 
physeqB.flt2 = filter_taxa(physeqB.flt2 , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 

taxtable.flt2 <- tax_table(physeqB.flt2)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqB.flt2)$Treatment)

#otutable <- as.data.frame(phyloseq::otu_table(physeqB.flt2))
#otutable <- round(otutable)

#cond <- sample_data(physeqB.flt2)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/x.all")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/x.all")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/aldex_significant_features.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/aldex_significant_features.csv")
#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
#  kable_styling(latex_options="scale_down")

## filter the genera whose abundances were significantly different in at least two of the treatments vs NPKS
aldex_significant_features <- rbind(aldex_significant_features0, aldex_significant_features1, aldex_significant_features2)
aldex_significant_features_presenttwice <- aldex_significant_features %>% 
  group_by(Genus) %>% 
  filter(n()>1)

## plot the genera whose abundances were significantly different in at least two of the treatments vs NPKS
physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments %in% c("Control", "NPKS","Poultry litter/N", "Poultry litter","NPKS/straw"), physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
#agglomerate to genus level
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "compositional")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)

subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% as.vector(aldex_significant_features_presenttwice$Genus) )
subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Phylum", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances", ncol = 3) + ggtitle("Genera different to NPKS in at least two other treatments (Aldex p < 0.05)") + theme(axis.text.x = element_text(angle = 45, hjust = 1))



```


```{r ancom, eval=FALSE, echo=FALSE, fig.width=9, fig.asp = 0.8, message=FALSE, warning=FALSE, cache=TRUE}

#compare with ancom
#export for qiime2 for ancom
#library(biomformat)
#otu <- as(otu_table(physeqB.flt2),"matrix")
#otu_biom <- make_biom(data=otu)
#write_biom(otu_biom,"~/Documents/Study/LaTrobe/Research/phD/Juan_project/ancom/otu_biom.biom")

#export metadata
#metadf <- data.frame(sample_data(physeqB.flt2)[,1:4]) %>% rownames_to_column("#SampleID")
#write.table(metadf,"~/Documents/Study/LaTrobe/Research/phD/Juan_project/ancom/sample-metadata.txt", sep="\t", row.names=FALSE, col.names=TRUE, quote=FALSE)
#...
require(tidyverse)
ancom <- readr::read_tsv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/ancom_NPKvsPL/ancom.tsv")%>% rename(Id = X1)
ancom_significant <- ancom  %>%  inner_join(taxtable.flt2) %>% dplyr::select(Id, Phylum, Genus, W, `Reject null hypothesis`) %>% filter(W >= 150)


physeqB.flt2 <- prune_samples(sample_names(physeqB) != "Juan07", physeqB)
physeqB.flt2 <- prune_samples(sample_data(physeqB.flt2)$Treatments %in% c("Control", "NPKS","Poultry litter/N", "Poultry litter","NPKS/straw"), physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) != 0, physeqB.flt2)
physeqB.flt2 = prune_taxa(taxa_sums(physeqB.flt2) >= 10, physeqB.flt2) #minimum reads per feature
#agglomerate to genus level
physeqB.flt2 <- tax_glom(physeqB.flt2 , "Genus")
physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "compositional")
physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)

subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% as.vector(ancom_significant$Genus) )

subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", facet.by = c("Genus"), fill = "Phylum", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + ggtitle("Ancom (top listed)")
```



# Pathway potentials - differences by treatment
Go [here](https://metacyc.org/pwy-search.shtml) to search for metacyc pathways
```{r aldex_pathways, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 3.5, message=FALSE, warning=FALSE, cache=TRUE}
# Aldex

#Sodosol NPK vs PL 
require(ALDEx2)
physeqpw.flt <- prune_samples(sample_data(physeq.pw)$Soiltypes == "Sodosol", physeq.pw)
physeqpw.flt <- prune_samples(sample_data(physeqpw.flt)$Treatments %in% c("NPKS","Poultry litter"), physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) != 0, physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) >= 10, physeqpw.flt) #minimum reads per feature
physeqpw.flt = filter_taxa(physeqpw.flt , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 
taxtable.flt2 <- tax_table(physeqpw.flt)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqpw.flt)$Treatment)

otutable <- as.data.frame(phyloseq::otu_table(physeqpw.flt))
otutable <- round(otutable)

#cond <- sample_data(physeqpw.flt)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/x.all_PW")
x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/x.all_PW")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/aldex_significant_features_PW.csv")
aldex_significant_features0 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKvsPL/aldex_significant_features_PW.csv")  %>% arrange(desc(effect)) %>% dplyr::select(-X)

#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
 # kable_styling(latex_options="scale_down")

#check them out (aldex or ancome)
#physeqB.flt2.clr <- microbiome::transform(physeqB.flt2, "compositional")
#physeqB.flt2.mlt.clr <- psmelt(physeqB.flt2.clr)
#subs <- physeqB.flt2.mlt.clr %>% dplyr::filter(Genus %in% as.vector(aldex_significant_features$Genus) )
#subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", facet.by = c("Genus") , fill = "Phylum", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances") + ggtitle("Aldex (p corrected < 0.02)")



#Sodosol NPK vs Straw+NPKS 
require(ALDEx2)
physeqpw.flt <- prune_samples(sample_data(physeq.pw)$Soiltypes == "Sodosol", physeq.pw)
physeqpw.flt <- prune_samples(sample_data(physeqpw.flt)$Treatments %in% c("NPKS","NPKS/straw"), physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) != 0, physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) >= 10, physeqpw.flt) #minimum reads per feature
physeqpw.flt = filter_taxa(physeqpw.flt , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 
taxtable.flt2 <- tax_table(physeqpw.flt)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqpw.flt)$Treatment)

otutable <- as.data.frame(phyloseq::otu_table(physeqpw.flt))
otutable <- round(otutable)

#cond <- sample_data(physeqpw.flt)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/x.all_PW")
x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/x.all_PW")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/aldex_significant_features_PW.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsStrawNPKS/aldex_significant_features_PW.csv")  %>% arrange(desc(effect)) %>% dplyr::select(-X)

#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
#  kable_styling(latex_options="scale_down")





#Sodosol NPK vs PL/N
require(ALDEx2)
physeqpw.flt <- prune_samples(sample_data(physeq.pw)$Soiltypes == "Sodosol", physeq.pw)
physeqpw.flt <- prune_samples(sample_data(physeqpw.flt)$Treatments %in% c("NPKS","Poultry litter/N"), physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) != 0, physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) >= 10, physeqpw.flt) #minimum reads per feature
physeqpw.flt = filter_taxa(physeqpw.flt , function(x) sum(x > 0) > (0.17*length(x)), TRUE) 
taxtable.flt2 <- tax_table(physeqpw.flt)@.Data %>% as.data.frame() %>% rownames_to_column("Id")

summary(sample_data(physeqpw.flt)$Treatment)

otutable <- as.data.frame(phyloseq::otu_table(physeqpw.flt))
otutable <- round(otutable)

#cond <- sample_data(physeqpw.flt)$Treatment
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/x.all_PW")
x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/x.all_PW")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.flt2) %>% dplyr::select(Id, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.1)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/aldex_significant_features_PW.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Juan_project/aldex_NPKSvsPL_N/aldex_significant_features_PW.csv") %>% dplyr::select(-X)
#kable(aldex_significant_features %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% 
#  kable_styling(latex_options="scale_down")

## filter the genera whose abundances were significantly different in at least two of the treatments vs NPKS
aldex_significant_features <- rbind(aldex_significant_features0, aldex_significant_features1, aldex_significant_features2)
aldex_significant_features_presenttwice <- aldex_significant_features %>% 
  group_by(Id) %>% 
  filter(n()>1) %>% filter(we.eBH <= 0.00025)

## plot the genera whose abundances were significantly different in at least two of the treatments vs NPKS

physeqpw.flt <- prune_samples(sample_data(physeq.pw)$Treatments %in% c( "Control","NPKS","Poultry litter/N", "Poultry litter","NPKS/straw"), physeq.pw)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) != 0, physeqpw.flt)
physeqpw.flt = prune_taxa(taxa_sums(physeqpw.flt) >= 10, physeqpw.flt) #minimum reads per feature
#agglomerate to genus level
physeqpw.flt.clr <- microbiome::transform(physeqpw.flt, "compositional")
physeqpw.flt.mlt.clr <- psmelt(physeqpw.flt.clr)

subs <- physeqpw.flt.mlt.clr %>% dplyr::filter(OTU %in% as.vector(aldex_significant_features_presenttwice$Id) )
subs  %>% ggbarplot(data = ., x = "Treatments", y = "Abundance", add = "mean_se", facet.by = c("OTU") , fill = "CLASS1", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free", ylab = "Relative abundances", ncol = 3) + ggtitle("Pathways different to NPKS in at least two treatments (Aldex p <= 0.00025)") + theme(axis.text.x = element_text(angle = 45, hjust = 1))



```




